tasks/Task.cpp

```diff
@@ -301,13 +341,135 @@ base::samples::RigidBodyState Task::convertToRbs(vpHomogeneousMatrix pose)
     rbs.position[0] = pose.getTranslationVector()[0];
     rbs.position[1] = pose.getTranslationVector()[1];
     rbs.position[2] = pose.getTranslationVector()[2];
-    
+
     vpRotationMatrix R;
     pose.extract(R);
-    
     vpQuaternionVector q(R);
-    
     rbs.orientation = base::Orientation(q.w(), q.x(), q.y(), q.z());
 
     return rbs;
 }
+
+void Task::transformInput(base::LinearAngular6DCommand cmd_in_body)
+{
+    //initialize setpoint on the body frame
+    //since the non-expected input are NaN, set it to default values
+    //in order to have valid values for the HomogeneusMatrix creation
+    for (int i = 0; i < 3; ++i)
+    {
+        if (!expected_inputs.linear[i]) {cmd_in_body.linear[i] = 0;}
+        if (!expected_inputs.angular[i])
+        {
+            if ( i == 0 || i == 1)
+            {
+                cmd_in_body.angular[i] = -M_PI_2;
```

Comment:
Not sure If I understood, because I think I'm following the suggested workflow:
> input_values -> normalized_input_values -> transformation -> homogeneous matrix.

_Input values_ : the argument "setpoint"-> 
_normalized_input_values_: the "_for_" loop -> 
_transformation_: building the homogeneus matrix from the setpoint (bdMo)and multiplying by the transformation matrix (cMb) -> 
_homogeneus matrix_: the result of the transformation.

Am I missing something?
> Having two different code paths for the NaN and normal cases spells trouble. In general, the less code paths there is, the better.

I don't see how to have only a single path in this case. Any suggestions?